manifests/unit_file.pp

```diff
@@ -90,6 +90,14 @@
 
   assert_type(Systemd::Unit, $name)
 
+  if $hasrestart {
+    warning("systemd::unit_file - ${title}: hasrestart is deprecated and will be removed in Version 4 of the module")
+  }
+
+  if $hasstatus == false {
```

Comment:
Should we actually make it `Optional[Boolean]`? The default of `hasstatus` in `service` is `true` anyway and it allows you to find out if it was passed in explicitly, which is a problem.

Reply:
good idea, implemented it.